src/distilabel/steps/base.py

```diff
@@ -48,7 +41,7 @@
 method defined in each `Step`"""
 
 
-class _Step(BaseModel, _Serializable, ABC):
+class _Step(RuntimeParametersMixin, _Serializable, ABC):
```

Comment:
Not sure either about "obscuring" the `BaseModel` inheritance under a mixin, that just feels off, since the mixin should be built on top but as a plug in / out component, not as a core component IMO, but can be wrong